modules/tinymce/src/core/main/ts/api/EditorUpload.ts

```diff
@@ -186,6 +158,10 @@ const EditorUpload = (editor: Editor): EditorUpload => {
           });
         }
 
+        if (shouldDispatchChange) {
```

Comment:
Why was this moved? This means the `undoManager.transact()` logic runs first which could alter the behaviour for integrators.

Reply:
I changed it back, I moved it in order to try to preserve the old behavior of change events when there is an error, I'll write more details here: https://github.com/tinymce/tinymce/pull/7840#discussion_r895247193